### DISCUSSION TITLE
mui provider works well in development mode but fail after build

### DISCUSSION THREAD

**Discussion by bbsweb** [facebook/docusaurus#7498] [Q&A] [answered]
I want to use docusaurus with mui components, and i add mui provider in docusaurus  layout.

```tsx
export default function ({ children }) {
  const { colorMode } = useColorMode();
  const theme = useMemo(
    () =>
      createTheme({
        palette: { mode: colorMode }
      }),
    [colorMode]
  );

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}
```

simple example: https://codesandbox.io/s/cranky-frog-oft12l

It works well in development mode. (mui components will follow docusaurus  color mode (dark or light))

But after build (`npm run build && npm run serve`), It cannot work.

![localhost_3000_](https://user-images.githubusercontent.com/20337958/170418057-d1fbf29d-efe6-42d0-bf70-d650294cf8df.png)

The mui components not follow the docusaurus  color mode.

It is so strange. 

Can anyone help me? Thanks a lot

**Comment by slorber** [comment, marked as answer]
We don't have good support for SSR and CSS-in-JS libs unfortunately, so problems with libs like MUI are expected. 

See also 
https://github.com/facebook/docusaurus/issues/3236
https://github.com/facebook/docusaurus/issues/5475

**Reply by bbsweb** [reply]
get it.  thanks
